tools/clang/lib/AST/SelectorLocationsKind.cpp

```diff
@@ -28,15 +28,15 @@ static SourceLocation getStandardSelLoc(unsigned Index,
     if (EndLoc.isInvalid())
       return SourceLocation();
     IdentifierInfo *II = Sel.getIdentifierInfoForSlot(0);
-    unsigned Len = II ? II->getLength() : 0;
+    int Len = II ? II->getLength() : 0;
```

Comment:
Probably fine but this is effectively cutting the container size in half right? Will issues arise if `II->getLength()` ever returns `signed int max + 1`?

Reply:
Yah, your observation is correct. But the call to getLocWithOffset below takes an int anyways. And my understanding is that in practice this length is never going to be very large.  

If that truly was a concern, then we'd also want to change getLocWithOffset to take something with a larger range than an int. Meaning I think that 'bug' already existed. The modern version of this file also still has the same thing: https://github.com/llvm/llvm-project/blob/2f9eee849f7d7414c447d26625517edc34437138/clang/lib/AST/SelectorLocationsKind.cpp#L31

Based on that, my take that it wasn't worth worrying about. Feel free to disagree!